lib/msal-common/src/cache/CacheManager.ts

```diff
@@ -390,7 +390,8 @@ export abstract class CacheManager implements ICacheManager {
                 return;
             }
 
-            if (!!homeAccountId && !this.matchHomeAccountId(entity, homeAccountId)) {
+            // homeAccountId can undefined, and we want to filter out cached items that have a homeAccountId of ""
+            if ((typeof homeAccountId === "string") && !this.matchHomeAccountId(entity, homeAccountId)) {
```

Comment:
Why are we checking "type"? Are we expecting data corruption? and should we check for "null" or "empty string" or both?

Reply:
@sameerag homeAccountId can be undefined or empty string. If it's undefined, we're skipping this check. Otherwise we need to account for a client credential application which has a homeAccountId of empty string.